src/libsystemd-network/ndisc-router.c

```diff
@@ -717,7 +717,7 @@ int sd_ndisc_router_dnssl_get_domains(sd_ndisc_router *rt, char ***ret) {
                                 e[n] = 0;
                                 r = dns_name_normalize(e, 0, &normalized);
                                 if (r < 0)
-                                        return r;
+                                        return r == -ENOMEM ? -ENOMEM : -EBADMSG;
```

Comment:
maybe add debug logging here, logging the original error code here first?

Reply:
i mean, it might be nice so that the original error code is surfaced before we convert it into a more generic one.